knip.jsonc

```diff
@@ -0,0 +1,46 @@
+{
+    "workspaces": {
+        ".": {
+            // These entries are complementary to the ones found in package.json
+            "entry": [
+                "lib/rules/index.js",
+                "tools/internal-rules/index.js",
```

Comment:
[Question] I was surprised to see these ad-hoc internal rules. Their existence feels to me like a justification to either build their functionality into ESLint core or into a plugin such as `eslint-plugin-eslint-plugin`. Maybe there should be tracking issues for that?

* `multiline-comment-style`: An option to be added to the rule? (or, maybe ESLint can drop banner comments?)
* `no-invalid-meta`: A configurable rule to be added to `eslint-plugin-eslint-plugin`?

_(probably a question for a team member)_

Reply:
In the past years, I actually moved as much of `internal-rules/` as I could (anything of generic value) into `eslint-plugin-eslint-plugin`. In particular, `no-invalid-meta` is only a skeleton of its former self and pretty much all of it is likely already handled by eslint-plugin-eslint-plugin's `require-meta-*` rules. So it's possible most or all of `internal-rules/` can be deleted...

Reply:
Let's start a separate discussion around this topic so we can keep this PR focused on Knip.

Reply:
For `multiline-comment-style`: #18122

For `no-invalid-meta`: https://github.com/eslint-community/eslint-plugin-eslint-plugin/issues/446

<details>
<summary>Issue text I drafted for this repo before noting that it'd be blocked on a rule not yet existing</summary>

Splitting out of https://github.com/eslint/eslint/pull/18005#discussion_r1458066983:

> In the past years, I actually moved as much of `internal-rules/` as I could (anything of generic value) into `eslint-plugin-eslint-plugin`. In particular, `no-invalid-meta` is only a skeleton of its former self and pretty much all of it is likely already handled by eslint-plugin-eslint-plugin's `require-meta-*` rules.

+1 from me. I think in general, most or all practices that ESLint would want to enforce are likely also things that should be made available in a standard ecosystem plugin like `eslint-plugin-eslint-plugin`.

It looks to me that [`tools/internal-rules/no-invalid-meta.js`](https://github.com/eslint/eslint/blob/3ae0574fc78c4a2b406625e4792cb2859cb9bcb1/tools/internal-rules/no-invalid-meta.js) enforces `meta.docs.recommended` exists. The `recommended` property is common in the ecosystem, though it isn't covered by any rules that exist right now in eslint-plugin-eslint-plugin. Filed 

</details>